xdsl/dialects/riscv_scf.py

```diff
@@ -240,6 +240,75 @@ def verify_(self):
                     f" got {block_arg.type}"
                 )
 
+    def print(self, printer: Printer):
+        printer.print_string(" (")
+        block_args = self.before_region.block.args
+        printer.print_list(
+            zip(block_args, self.arguments, strict=True),
+            lambda pair: printer.print(pair[0], " = ", pair[1]),
+        )
+        printer.print_string(") : ")
+        printer.print_operation_type(self)
+        printer.print_string(" ")
+        printer.print_region(self.before_region, print_entry_block_args=False)
+        printer.print(" do ")
+        printer.print_region(self.after_region)
+        if self.attributes:
+            printer.print_op_attributes(self.attributes, print_keyword=True)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        def parse_assignment():
+            arg = parser.parse_argument(expect_type=False)
+            parser.parse_punctuation("=")
+            operand = parser.parse_unresolved_operand()
+            return arg, operand
+
+        tuples = parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN,
+            parse_assignment,
+        )
+
+        parser.parse_punctuation(":")
+        type_pos = parser.pos
+        function_type = parser.parse_function_type()
+
+        if len(tuples) != len(function_type.inputs.data):
+            parser.raise_error(
```

Comment:
@math-fehr what's the best way to test this?

Reply:
I guess a filecheck?

Reply:
I tried it and it doesn't work, the whole thing fails as opposed to successfully finding the string with the error

Reply:
`--parsing-diagnostics`